### DISCUSSION TITLE
How can I run mini.starter when opening a directory?

### DISCUSSION THREAD

**Discussion by andradei** [nvim-mini/mini.nvim#746] [Q&A] [answered]
When I run `nvim .` or `nvim some/dir/`, how can I open `mini.starter` instead of netrw?

**Comment by echasnovski** [comment, marked as answer]
Unfortunately, this is not as easy as it might seem. I learnt it the hard way in 'mini.files'.

Basically, there are two steps to achieve this:
- Disable 'netrw' from taking over when directory is edited.
- Create an autocommand for `BufEnter` event to decide whether a buffer is for directory and if it is, open Starter buffer.

The most robust way to do this comes from 'mini.files' ([this part](https://github.com/echasnovski/mini.nvim/blob/d7d079d6f99c95218d6059e0eaf83997990b2e0e/lua/mini/files.lua#L1141-L1146) and [this part](https://github.com/echasnovski/mini.nvim/blob/d7d079d6f99c95218d6059e0eaf83997990b2e0e/lua/mini/files.lua#L1182-L1200). And for the case of 'mini.starter' it can be done with something like this:

```lua
-- Stop 'netrw' from taking over when directory path is edited
vim.cmd('silent! autocmd! FileExplorer *')
vim.cmd('autocmd VimEnter * ++once silent! autocmd! FileExplorer *')

-- Open Starter buffer when directory path is edited
local open_starter_on_dir = function(data)
  -- Make early returns
  if vim.api.nvim_get_current_buf() ~= data.buf then return end
  if vim.b.ministarter_processed_dir then return vim.api.nvim_buf_delete(data.buf, { force = true }) end
  local path = vim.api.nvim_buf_get_name(0)
  if vim.fn.isdirectory(path) ~= 1 then return end

  -- Show starter buffer
  vim.bo.bufhidden = 'wipe'
  vim.b.ministarter_processed_dir = true
  vim.schedule(MiniStarter.open)
end
vim.api.nvim_create_autocmd('BufEnter', { callback = open_starter_on_dir })
```

All parts with `vim.b.ministarter_processed_dir` is for to be able to press `<C-c>` and close Starter buffer without retriggering it again (as `BufEnter` event for the previous directory buffer will be triggered).

**Reply by andradei** [reply]
Thank you for the detailed info!
